include/Core/Config.php

```diff
@@ -132,15 +132,33 @@ public static function set($family,$key,$value) {
 		$dbvalue = (is_array($value)?serialize($value):$value);
 		$dbvalue = (is_bool($dbvalue) ? intval($dbvalue) : $dbvalue);
 
-		$ret = q("INSERT INTO `config` ( `cat`, `k`, `v` ) VALUES ( '%s', '%s', '%s' )
-ON DUPLICATE KEY UPDATE `v` = '%s'",
+		// The "INSERT" command is very cost intense. It saves performance to do it this way.
```

Comment:
The single query allows for an atomic transaction. Granted, in this case it isn't completely warranted since there's little chance that two users would set the same non-existing setting, resulting in an INSERT error for one of them, but it generally good practice.

I also believe that performances when editing data is not a crucial factor. When retrieving data, most definitely, but I will value data consistency on write over raw performance since it is not expected by users, even unconsciously.